modules/wysiwyg.php

```diff
@@ -19,15 +34,20 @@ function wysiwyg_d7csp_hosts() {
   $css_themes[] = variable_get('admin_theme');
   $css_themes[] = variable_get('theme_default', 'bartik');
   $css_themes = array_unique(array_filter($css_themes));
+  $backup_active_theme = $GLOBALS['theme_key'];
   foreach ($css_themes as $theme_name) {
-    $theme = $themes[$theme_name];
-    if (file_exists($template_php = dirname($theme->filename) . '/template.php')) {
-      require_once $template_php;
-      $alter_fn = "{$theme->name}_d7csp_hosts_alter";
-      if (function_exists($alter_fn)) {
-        $alter_fn($hosts);
```

Comment:
I'd prefer to replace the loop with something simpler like this instead of trying to find a d7csp hook:
```php
foreach ($css_themes as $theme_name) {
  $css_path = theme_get_setting('foundation_assets_css', $css_theme)
  if ($css_path && $host = parse_url($css_path, PHP_URL_HOST)) {
    $hosts['style-src'][] = $host;
    $hosts['font-src'][] = $host;
    $hosts['img-src'][] = $host;
  }
}
```

But your code also works, so fine with me to not invest even more time.